### PR TITLE
fixes for video and other minor fixes

### PR DESCRIPTION
**Pull request by hardening**

### CODE REVIEW

File: libfreerdp/gdi/video.c [hunk 1]
```diff
@@ -140,9 +140,9 @@ void gdi_video_control_uninit(rdpGdi* gdi, VideoClientContext* video)
 	gdi->video = NULL;
 }
 
-static void gdi_video_timer(void* ctx, TimerEventArgs* timer)
+static void gdi_video_timer(rdpContext *ctx, TimerEventArgs* timer)
```
Comment (akallabeth) [inline libfreerdp/gdi/video.c]: prototype of event handler uses `void*` so better keep that

File: libfreerdp/gdi/video.c [hunk 2]
```diff
@@ -140,20 +140,20 @@ void gdi_video_control_uninit(rdpGdi* gdi, VideoClientContext* video)
...
+	rdpGdi* gdi = ctx->gdi;
 
 	if (gdi && gdi->video)
 		gdi->video->timer(gdi->video, timer->now);
 }
 
 void gdi_video_data_init(rdpGdi* gdi, VideoClientContext* video)
 {
-	PubSub_SubscribeTimer(gdi->context->pubSub, gdi_video_timer);
+	PubSub_SubscribeTimer(gdi->context->pubSub, (pTimerEventHandler)gdi_video_timer);
```
Comment (akallabeth) [inline libfreerdp/gdi/video.c]: These casts are quite bad as the compiler will no longer warn if something in the `PubSub_SubscribeTimer` arguments changes

File: libfreerdp/gdi/video.c [hunk 3]
```diff
@@ -140,20 +140,20 @@ void gdi_video_control_uninit(rdpGdi* gdi, VideoClientContext* video)
...
 }
 
 void gdi_video_data_uninit(rdpGdi* gdi, VideoClientContext* context)
 {
-	PubSub_UnsubscribeTimer(gdi->context->pubSub, gdi_video_timer);
+	PubSub_UnsubscribeTimer(gdi->context->pubSub, (pTimerEventHandler)gdi_video_timer);
```
Comment (akallabeth) [inline libfreerdp/gdi/video.c]: same as above

File: libfreerdp/gdi/video.c (hunk 2 above)
Comment (hardening) [reply libfreerdp/gdi/video.c]: Well I will change them, but nothing can change in the arguments as these functions are automatically generated by some macros.

**Comment by akallabeth** [comment]
@hardening I've created d119e5222db771e28a5f09bc282eb6d1b3d1218a to remove all the casts from pubsub related stuff. Should help if there are changes we at least get some warnings.

**Pull request merged**